### PR TITLE
Refactor core functionalities

### PR DESCRIPTION
**Pull request by imbrn**
- [x] Refactor `core` functions to decouple modifiers
- [x] Add `some` modifier;
- [x] Add `every` modifier
- [x] Update tests

### CODE REVIEW

File: src/v8n.js [hunk 1]
```diff
@@ -68,7 +69,8 @@ function v8n() {
    * @module Validation
    */
   const context = {
-    chain: []
+    chain: [],
```
[the remark this answers is not in the document]
Comment (imbrn) [reply src/v8n.js]: I'm going to change `_modifiers` to modifiers as renaming `chain` is going to break the public API.

[the remark this answers is not in the document]
Comment (sebastianbarfurth) [reply src/v8n.js]: Is chain really public API? Seems like a backend detail.

File: src/v8n.js [hunk 2]
```diff
@@ -384,8 +445,21 @@ const modifiers = {
    * v8n()
    *  .not.equal("three");
    */
-  not() {
-    this.invert = true;
+  not: {
```
[the remark this answers is not in the document]
Comment (imbrn) [reply src/v8n.js]: Agree!

File: src/v8n.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (imbrn) [reply src/v8n.js]: Yeah. The context is exposed by the v8n function so are its properties.

[the remark this answers is not in the document]
Comment (sebastianbarfurth) [reply src/v8n.js]: Seems like they shouldn't be to be honest. Possibly deprecate access for 2.0.0.

[the remark this answers is not in the document]
Comment (imbrn) [reply src/v8n.js]: I think that too

**Pull request merged**